app/schema/widgets/date_widget.py

```diff
@@ -10,85 +10,90 @@
-                'selected': (selected_day == day),
-                'disabled': False,
-            }))
-        return days
-
-    def _get_months(self, selected_month):
-        if selected_month:
-            selected_month = int(selected_month)
-
-        months = []
-        for month in range(1, 13):
-            months.append(ObjectFromDict({
-                'value': month,
-                'text': calendar.month_name[month],
-                'selected': (selected_month == month),
-                'disabled': False,
-            }))
-        return months
 
     def render(self, answer_state):
+
         if answer_state.input:
             parts = answer_state.input.split('/')
         else:
             parts = [None, None, '']
 
-        widget_params = {
-            'legend': answer_state.schema_item.label,
-            'fields': {
-                'day': {
-                  'label': {
-                    'for': self.name + '-day',
-                    'text': 'Day',
+        widget_params = {}
+        widget_params['legend'] = answer_state.schema_item.label
+        widget_params['fields'] = self._get_date_fields(parts)
+
+        return render_template('partials/widgets/date_widget.html', **widget_params)
+
+    def _get_date_fields(self, parts):
+
+        fields = {}
+        fields['day'] = self._get_day_field(parts[0])
+        fields['month'] = self._get_month_field(parts[1])
+        fields['year'] = self._get_year_field(parts[2])
+
+        return fields
+
+    def _get_day_field(self, day):
+        day = {
+              'label': {
+                  'for': self.name + '-day',
+                  'text': 'Day',
                   },
-                  'input': {
-                    'type': 'text',
-                    'value': parts[0],
-                    'placeholder': 'DD',
-                    'name': self.name + '-day',
-                    'id': self.name + '-day',
+              'input': {
+                  'type': 'text',
+                  'value': day,
+                  'placeholder': 'DD',
+                  'name': self.name + '-day',
+                  'id': self.name + '-day',
                   },
-                },
-                'month': {
-                  'label': {
+                }
+        return day
+
+    def _get_month_field(self, month):
+
+        month = {
+                'label': {
                     'for': self.name + '-month',
                     'text': 'Month',
-                  },
-                  'select': {
-                    'options': self._get_months(parts[1]),
+                },
+                'select': {
+                    'options': self._get_months(month),
                     'name': self.name + '-month',
                     'id': self.name + '-month',
                   },
-                },
-                'year': {
-                  'label': {
+                }
+        return month
+
+    def _get_year_field(self, year):
+        year = {
```

Comment:
inline return would be nicer than reassigning year to be something of a totally different type